scripts/prepdocslib/filestrategy.py

```diff
@@ -25,8 +26,10 @@ def __init__(
         self,
         list_file_strategy: ListFileStrategy,
         blob_manager: BlobManager,
+        json_parser: JsonParser,
         pdf_parser: PdfParser,
-        text_splitter: TextSplitter,
+        pdf_text_splitter: PdfTextSplitter,
```

Comment:
Now that there are more parsers and splitters, I was wondering if you have thoughts about how to avoid having to pass in the classes? Imagine we also had parsers for HTML, CSV, etc, this would involve more parameters to init and attributes. Ideally we could avoid that while still allowing flexibility of using different parsers.

Reply:
yeah, good call. doing a refactor